src/main/tut/kleisli-categories.md

```diff
@@ -0,0 +1,71 @@
+```Haskell
+type Writer a = (a, String)
+```
+```tut:silent
+type Writer[A] = (A, String)
+```
+................
+```Haskell
+a -> Writer b
+```
+```scala
+A => Writer[B]
+```
+................
+```Haskell
+(>=>) :: (a -> Writer b) -> (b -> Writer c) -> (a -> Writer c)
+```
+```scala
+def >=>[A, B, C](m1: A => Writer[B], m2: B => Writer[C]): A => Writer[C]
+```
+................
+```Haskell
+m1 >=> m2 = \x -> 
+    let (y, s1) = m1 x
+        (z, s2) = m2 y
+    in (z, s1 ++ s2)
+```
+```tut:silent
+object kleisli {
+  implicit class Fishy[A, B](m1: A => Writer[B]) {
+   def >=>[C](m2: B => Writer[C]): A => Writer[C] =
+    x => {
+      val (y, s1) = this.m1(x)
+      val (z, s2) = m2(y)
+      (z, s1 + s2)
+    }
+  }
+}
+```
+................
+```Haskell
+return :: a -> Writer a
+return x = (x, "")
+```
+```tut:silent
+def `return`[A](x: A): Writer[A] = (x, "")
+```
```

Comment:
Again, interesting point. I'd be inclined to call it `pure` and add a comment, but not 100% sure

Reply:
+1 on calling it `pure`